src/app/clusters/administrator-commissioning-server/administrator-commissioning-server.cpp

```diff
@@ -0,0 +1,85 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/****************************************************************************
+ * @file
+ * @brief Implementation for the Administrator Commissioning Cluster
+ ***************************************************************************/
+
+#include <app/CommandHandler.h>
+#include <app/server/Server.h>
+#include <app/util/af.h>
+#include <support/CodeUtils.h>
+#include <support/logging/CHIPLogging.h>
+
+using namespace chip;
+
+bool emberAfAdministratorCommissioningClusterOpenCommissioningWindowCallback(chip::EndpointId endpoint,
+                                                                             app::CommandHandler * commandObj,
+                                                                             uint16_t commissioningTimeout, ByteSpan pakeVerifier,
```

Comment:
Please verify discriminator <= 4095, e.g. ((1u << 12) - 1u), which is the valid discriminator bounds.